dbms/src/Interpreters/Join.cpp

```diff
@@ -793,20 +793,45 @@ void mergeNullAndFilterResult(
     auto [filter_vec, nullmap_vec] = getDataAndNullMapVectorFromFilterColumn(current_filter_column);
     if (nullmap_vec != nullptr)
     {
-        for (size_t i = 0; i < nullmap_vec->size(); ++i)
+        if (filter_column.empty())
         {
-            if (filter_column[i] == 0)
-                continue;
-            if ((*nullmap_vec)[i])
-                filter_column[i] = null_as_true;
+            filter_column.insert(nullmap_vec->begin(), nullmap_vec->end());
+            if (null_as_true)
+            {
+                for (size_t i = 0; i < nullmap_vec->size(); ++i)
+                    filter_column[i] = filter_column[i] || (*filter_vec)[i];
+            }
             else
-                filter_column[i] = filter_column[i] && (*filter_vec)[i];
+            {
+                for (size_t i = 0; i < nullmap_vec->size(); ++i)
+                    filter_column[i] = !filter_column[i] && (*filter_vec)[i];
```

Comment:
Doesn't seem to be equivalent to https://github.com/pingcap/tiflash/blob/e597b78a6f5595ba50ffe839daca798ced1d1bc4/dbms/src/Interpreters/Join.cpp#L798-L803

Reply:
sorry, my mistake..